src/components/pages/MainQuizPage/EndQuizButton/EndQuizButton.tsx

```diff
@@ -1,8 +1,29 @@
+import { useState } from "react";
 import { StyledButton } from "../../../spreadedComponents/StyledButton/StyledButton";
+import { createPortal } from "react-dom";
+import { ModalPage } from "../../ModalPage/ModalPage";
+import { START } from "../../../Router/routes";
+import { useNavigate } from "react-router-dom";
 
 const EndQuizButton = () => {
+
+    const [showModal, setShowModal] = useState(false);
+    const navigate = useNavigate();
+
     return (
-        <StyledButton onClickHandler={() => { }} text="Finish test" />
+        <>
+            <StyledButton onClickHandler={() => setShowModal(true)} text="Finish test" />
+            {
+                showModal && createPortal(
+                    <ModalPage
+                        setShowModal={() => setShowModal(false)}
+                        navigateToResult={() => navigate(START)}
```

Comment:
please move these callbacks into constants because every time when you change `showModal` value you pass new callbacks
this is bad from performance perspective